mux_circuits/src/bitshift.rs

```diff
@@ -23,7 +36,8 @@ fn ilog2_rounded_up(x: u16) -> u32 {
 ///   be specified MSB first.
 /// * `right` - Whether to shift right or left.
 /// * `zeros` - Whether to fill the shifted bits with zeros.
-pub fn bitshift(inputs: u16, shift_size: u16, right: bool, zeros: bool) -> MuxCircuit {
+/// * `arith` - Whether to do arithmetic shift (consider sign bit), requires right shift and not zeros.
+pub fn bitshift(inputs: u16, shift_size: u16, right: bool, mode: ShiftMode) -> MuxCircuit {
```

Comment:
Make the shift direction an enum as well so that we don't have boolean blindness in the direction either.